File: api-server/src/models/Skill.ts
import mongoose, { Schema, Document } from 'mongoose';

export interface ISkill extends Document {
  name: string;
  category: string;
  isActive: boolean;
  createdAt: Date;
}

const skillSchema = new Schema<ISkill>({
  name: { type: String, required: true },
  category: { type: String },
  isActive: { type: Boolean, default: true },
  createdAt: { type: Date, default: Date.now },
});

export const Skill = mongoose.model<ISkill>('Skill', skillSchema, 'skills');

File: doer_app/lib/data/models/skill.dart
/// Skill model fetched from API.
class Skill {
  final String id;
  final String name;
  final String category;
  final bool isActive;

  const Skill({
    required this.id,
    required this.name,
    required this.category,
    this.isActive = true,
  });

  factory Skill.fromJson(Map<String, dynamic> json) {
    return Skill(
      id: json['_id']?.toString() ?? json['id']?.toString() ?? '',
      name: json['name'] ?? '',
      category: json['category'] ?? '',
      isActive: json['isActive'] ?? json['is_active'] ?? true,
    );
  }
}
